File: 24-QUE/X- Is Subsequence/Sequence.java
import java.util.Scanner;

public class Sequence {
    public static boolean isSubsequence(String s, String t) {
        int n = s.length();
        int m = t.length();
        int i = 0;
        int j = 0;
        while (i < n && j < m) {
            if (s.charAt(i) != t.charAt(j)) {
                j++;
            } else {
                i++;
                j++;
            } 
            if (i == n) {
            return true;
            } 

            return true;

        }
        
        return false;
    }
    
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        String s = scanner.nextLine();
        String t = scanner.nextLine();
        System.out.println(isSubsequence(s, t));
    }
}

File: 24-QUE/X- Is Subsequence/Sequence.py
def isSubsequence(s,t):
    n=len(s)
    m=len(t)
    i=0
    j=0
    while i<n and j<m:
        if s[i]!=t[j]:
            j=j+1
        else:
            i=i+1
            j=j+1
    if i==n:
        return True
    return False

        
s=input()
t=input()
print(isSubsequence(s,t))